### PR TITLE
feat(2781): support gRPC reflection custom headers

### LINKED ISSUE #2781: feat: Add Support for Custom Headers in gRPC Reflection Requests
It would be beneficial to have a feature that allows users to add custom headers, methods and bodies to link resources. 

**DSL Proposal**

```graphql
schema
  #Currently only src is supported
  @link(src: "https://abc.com", type: Grpc)

  # Proposal is to support other HTTP fields in link
  @link(
    src: "https://abc.com"
    method: GET
    body: "..."
    headers: [{}]
    type: Grpc
  ) {
  query: Query
}
```

**Technical Requirements**
- Add unit & integration tests and integration test wherever requried.
- For `method` use the predefined `Method` type in config.
- For `headers` use the predefined `Vec<KeyValue>` type in config.
- Ensure a validation error is thrown if http fields are provided with a file path.
- Except src and type, all other fields are optional.

**Additional Requirements**
- Update the documentation on the website

### PR DESCRIPTION
**Pull request by beelchester**
fixes #2781
/claim #2781

### CODE REVIEW

File: src/core/proto_reader/fetch.rs [hunk 1]
```diff
@@ -69,14 +71,21 @@ struct ReflectionResponse {
     file_descriptor_response: Option<FileDescriptorProtoResponse>,
 }
 
+#[allow(dead_code)]
```
Comment (karatakis) [inline src/core/proto_reader/fetch.rs:74]: This is not needed, please remove it

File: src/core/proto_reader/fetch.rs [hunk 2]
```diff
@@ -69,14 +71,21 @@ struct ReflectionResponse {
...
 pub struct GrpcReflection {
     server_reflection_method: GrpcMethod,
     url: String,
+    headers: Option<Vec<KeyValue>>,
     target_runtime: TargetRuntime,
 }
 
+#[allow(clippy::too_many_arguments)]
```
Comment (karatakis) [inline src/core/proto_reader/fetch.rs:82]: This is not needed, please remove it

File: src/core/proto_reader/fetch.rs [hunk 3]
```diff
@@ -135,16 +145,28 @@ impl GrpcReflection {
             )
             .as_str(),
         );
+
+        let mut headers = vec![];
+        if let Some(custom_headers) = self.headers.clone() {
```
Comment (karatakis) [inline src/core/proto_reader/fetch.rs:150]: No need to `clone` use `&`

File: src/core/proto_reader/fetch.rs [hunk 4]
```diff
@@ -135,16 +145,28 @@ impl GrpcReflection {
...
+            for header in custom_headers {
+                headers.push((
+                    HeaderName::from_str(&header.key.clone())?,
```
Comment (karatakis) [inline src/core/proto_reader/fetch.rs:153]: No need to `clone`

File: src/core/proto_reader/reader.rs [hunk 5]
```diff
@@ -30,9 +31,18 @@ impl ProtoReader {
         Self { reader, runtime }
     }
 
+    #[allow(clippy::too_many_arguments)]
```
Comment (karatakis) [inline src/core/proto_reader/reader.rs:34]: Not needed, please remove

**Review by karatakis — COMMENTED** [review]
I found some stuff, but nothing scary

**Review by karatakis — APPROVED** [review]
Looks ok

**Comment by tusharmath** [comment]
@beelchester Can you please update the documentation for this change.

**Pull request merged**